pkg/client/events/replay_client.go

```diff
@@ -142,17 +144,19 @@ func (rClient *replayClient[T, R]) EventsSequence(ctx context.Context) R {
 // goRemapEventsSequence publishes events observed by the most recent cached
 // events type replay observable to the given publishCh
 func (rClient *replayClient[T, R]) goRemapEventsSequence(ctx context.Context, publishCh chan<- T) {
-	// Subscribe to the replayObsCache to listen for changes to the "active"
-	// replay observable.
-	cachedEventTypeObserver := rClient.replayObsCache.Subscribe(ctx)
-	for eventObs := range cachedEventTypeObserver.Ch() {
-		// Publish events from the new "active" replay observable to the given
-		// publish channel.
-		eventObserver := eventObs.Subscribe(ctx)
-		for event := range eventObserver.Ch() {
-			publishCh <- event
-		}
-	}
+	channel.ForEach[R](
+		ctx,
+		rClient.replayObsCache,
+		func(ctx context.Context, eventTypeObs R) {
+			// TODO_CONSIDERATION: hold a reference to the previous event type
+			// observable and call `#UnsubscribeAll()` here in the event that it
+			// somehow starts receiving notifications again.
```

Comment:
`somehow starts receiving notifications again.` 

Is this a bug you encountered?

Reply:
So long as we're using the gorilla websocket implementation and their comment about **all** receive errors being persistent, this should not be possible. However, if that assumption fails, due to the fact that obervables don't have a "closed" state (either has observers or not), a non-persistent error could cause the retry loop to trigger which would result in this scenario.

Reply:
Maybe #PUC with some deets from :point_up_2:

Reply:
Implemented instead